addons/mail/models/mail_thread.py

```diff
@@ -3988,6 +3989,20 @@ def message_change_thread(self, new_thread, new_parent_message=False):
         # other than comment: reset subtype
         msg_vals["subtype_id"] = None
         msg_not_comment.write(msg_vals)
+
+        # move the attachments
+        if move_attachments:
+            attachments = self.env['ir.attachment'].sudo().search([('res_model', '=', self._name), ('res_id', '=', self.id)])
+            attachments.write({'res_model': new_thread._name, 'res_id': new_thread.id})
+
+        # move the activities
+        if move_activities:
+            activities = self.env['mail.activity'].sudo().search([('res_model', '=', self._name), ('res_id', '=', self.id)])
+            activities.write({'res_model': new_thread._name, 'res_id': new_thread.id})
+            # reset activity types if they are incompatible with the new model
+            if self._name != new_thread._name:
+                activities.filtered('activity_type_id.res_model').activity_type_id = False
```

Comment:
I'm not sure if this is the right way to do it since this field is required

Reply:
You may call '_default_activity_type_for_model' that gives the default activity type linked to a model, from the most specific to the least specific one.

Reply:
Thanks for the feedback, I added that